On illumos LC_GLOBAL_LOCALE is not a constant, but a function call, so compiler doesn't like such usage

--- coreutils-8.22/gnulib-tests/test-locale.c.~1~	2015-01-15 18:47:13.223845265 +0300
+++ coreutils-8.22/gnulib-tests/test-locale.c	2015-01-15 18:49:11.794093717 +0300
@@ -33,11 +33,6 @@
     LC_TIME
   };
 
-#if HAVE_NEWLOCALE
-/* Check that the locale_t type and the LC_GLOBAL_LOCALE macro are defined.  */
-locale_t b = LC_GLOBAL_LOCALE;
-#endif
-
 /* Check that the 'struct lconv' type is defined.  */
 struct lconv l;
 int ls;
@@ -49,6 +44,14 @@
 int
 main ()
 {
+
+#if HAVE_NEWLOCALE
+/* Check that the locale_t type and the LC_GLOBAL_LOCALE macro are defined.  */
+  locale_t b;
+
+  b = LC_GLOBAL_LOCALE;
+#endif
+ 
   /* Check that 'struct lconv' has the ISO C and POSIX specified members.  */
   ls += sizeof (*l.decimal_point);
   ls += sizeof (*l.thousands_sep);
